### PR TITLE
Added instructions for the labyrinth challenge

### PR DESCRIPTION
**Pull request by AlphaGit**
I just added the instructions and polished up the draft readme that @danielisaacgeslin created.

This is a new challenge that we'll be sending out to React candidates.

There is no startup code in this repository (even though we have some) because we want Sr developers to start from scratch. Ssr developers (or lower) will get a startup folder to get them started.

(@danielisaacgeslin would you please review here too? I could not add you as a reviewer.)

### CODE REVIEW

**Comment by AlphaGit** [comment]
PS: I just noticed a few glitches in the rendered branch (which you can [see here](https://github.com/MakingSense/dev-challenges/blob/labyrinth-challenge/React/labyrinth-challenge.md)). I will adjust and push a fix quickly.

**Comment by andresmoschini** [comment]
@AlphaGit, the readme is very clear and motivating.

The challenge is also very interesting, great work @danielisaacgeslin!

Do you have an estimation of the required time to do this work? maybe having that information would be also useful for the applicants.

File: React/labyrinth-challenge.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+* You can add any library you consider that will help you build the game, as long as they are "regular web application" libraries. This means you can use libraries to help you with styles, state management, testing, etc. You cannot use libraries that are meant for building games.
+* You have to test the application / components, covering as much functionality as you can.
+* You have to build this application to the best of your abilities, considering customization and the possibility of extending it in the future.
+
+## Nice to have
+
+Not doing any of the following will **not** get you penalized, and you are **not** required to do any of them. If you decide to do any of them, they might add to your qualification, always considering the quality of the code you create.
+
+* Visual improvements (e.g., the circle/cells could you images, or the movement could be animated)
```
Comment (danielisaacgeslin) [inline React/labyrinth-challenge.md:29]: could `use` images?

**Comment by AlphaGit** [comment]
@andresmoschini 

> Do you have an estimation of the required time to do this work? maybe having that information would be also useful for the applicants.

@danielisaacgeslin tells me he had the working final solution running in about 2 hs. However, we all know how efficient he is so let's say that for a person approaching this problem anew, it could take them somewhere around 4hs or 5hs.

HR will be sending this challenge and communicating to people that they have 2 calendar days to provide a solution. This is considering especially those that don't have much time outside of their regular work hours.

I did not want to include that timing here because I wasn't sure if it would motivate people or put them under pressure, or maybe put them in a mentality of "I'll just do something quick and provide a solution". So... I'm not sure. Opinions?

File: React/labyrinth-challenge.md (hunk 1 above)
Comment (AlphaGit) [reply React/labyrinth-challenge.md]: Woops -- thanks! Fixed!

**Comment by danielisaacgeslin** [comment]
> @AlphaGit, the readme is very clear and motivating.
> 
> The challenge is also very interesting, great work @danielisaacgeslin!
> 
> Do you have an estimation of the required time to do this work? maybe having that information would be also useful for the applicants.

the basic version from scratch took me about 2hs and a half. I would say between 3hs and 5hs top is enough to solve (if candidate is able to solve it at all)

**Comment by andresmoschini** [comment]
> I did not want to include that timing here because I wasn't sure if it
would motivate people or put them under pressure, or maybe put them in a
mentality of "I'll just do something quick and provide a solution". So...
I'm not sure. Opinions?

You are right. It could probably take 10 hours or more to me. So, if you
said me that 5 hours should be enough I would feel unconformable hehe.

**Comment by AlphaGit** [comment]
@danielisaacgeslin I removed the cell numbers from the example gif. If you're ok with the current approach let me know and we can proceed to merge this PR.

If not, let me know what else you'd change and we can make it happen. :)

**Comment by danielisaacgeslin** [comment]
> @danielisaacgeslin I removed the cell numbers from the example gif. If you're ok with the current approach let me know and we can proceed to merge this PR.
> 
> If not, let me know what else you'd change and we can make it happen. :)

it looks perfect

**Pull request merged**